yt_dlp/extractor/callin.py

```diff
@@ -0,0 +1,123 @@
+        url = episode['m3u8']
+        formats = self._extract_m3u8_formats(url, display_id, ext='ts')
+        self._sort_formats(formats)
+
+        show = traverse_obj(episode, ('show', 'title'))
+        show_id = traverse_obj(episode, ('show', 'id'))
+
+        # get the show metadata for some supplementary info
+        show_json = None
+        app_slug = (
+            self._html_search_regex(
+                '<script\\s+src=["\']/_next/static/([a-zA-Z0-9_]+)/_',
+                webpage, 'app slug', fatal=False)
+            # this sometimes works, but sometimes seems to lag changes in the actual urls used.
+            or next_data.get('buildId')
+        )
+        show_slug = traverse_obj(episode, ('show', 'linkObj', 'resourceUrl'))
+        if app_slug and show_slug and '/' in show_slug:
+            show_slug = show_slug.rsplit('/', 1)[1]
+            show_json_url = f'https://www.callin.com/_next/data/{app_slug}/show/{show_slug}.json'
+            show_json = self._download_json(show_json_url, display_id, fatal=False)
+
+        # if we don't have the show metadata to tell us the host explicitly,
+        # guess that it's the first speaker listed for the episode
+        host = (traverse_obj(show_json, ('pageProps', 'show', 'hosts', 0))
+                or traverse_obj(episode, ('speakers', 0)))
+
+        host_nick = traverse_obj(host, ('linkObj', 'resourceUrl'))
+        host_nick = host_nick.rsplit('/', 1)[1] if (host_nick and '/' in host_nick) else None
+
+        cast = list(filter(None, [
+            self.try_get_user_name(u) for u in
+            traverse_obj(episode, (('speakers', 'callerTags'), ...)) or []
+        ]))
+
+        episode_list = traverse_obj(show_json, ('pageProps', 'show', 'episodes')) or []
+        episode_number = next(
+            (len(episode_list) - i for (i, e) in enumerate(episode_list) if e.get('id') == id),
+            None
+        )
+
+        return {
+            'id': id,
+            'display_id': display_id,
+            'title': title,
+            'formats': formats,
+            'thumbnail': traverse_obj(episode, ('show', 'photo')),
+            'description': episode.get('description'),
+            'uploader': self.try_get_user_name(host) if host else None,
+            'timestamp': episode.get('publishedAt'),
+            'uploader_id': host_nick,
+            'uploader_url': traverse_obj(show_json, ('pageProps', 'show', 'url')),
+            'channel': show,
+            'channel_id': show_id,
+            'channel_url': traverse_obj(episode, ('show', 'linkObj', 'resourceUrl')),
+            'duration': float_or_none(episode.get('runtime')),
+            'view_count': int_or_none(episode.get('plays')),
+            'categories': traverse_obj(episode, ('show', 'categorizations', ..., 'name')),
+            'cast': cast if cast else None,
+            'series': show,
+            'series_id': show_id,
```

Comment:
Why is channel and series the same?

Reply:
Callin just has "shows", which seem analogous to both. Should I use only one, and if so, which?

Reply:
I don't use the site. It's up to you to decide which one (or both) makes sense. I asked because it seems odd to me that show and channel can be same. Don't populate fields just for the sake of it. But if you think this makes sense in the context of that site, you can keep it. 

```
    channel:        Full name of the channel the video is uploaded on.
    series:         Title of the series or programme the video episode belongs to.
```

Reply:
I think it makes sense to use both; `series` is the preferred field for podcast episodes, but it's still useful to have the show url, and that only fits in `channel_url`. (Duplicating the name and id makes it clear that "series" and "channel" here refer to the same thing.)